server/command.go

```diff
@@ -240,6 +249,173 @@ func (p *Plugin) executeShowCommand(args *model.CommandArgs) (*model.CommandResp
 	return &model.CommandResponse{}, nil
 }
 
+func (p *Plugin) executeShowLinksCommand(args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	if !p.API.HasPermissionTo(args.UserId, model.PermissionManageSystem) {
+		return p.cmdError(args.UserId, args.ChannelId, "Unable to execute the command, only system admins have access to execute this command.")
+	}
+
+	links, err := p.store.GetLinks()
+	if err != nil {
+		p.API.LogDebug("Unable to get links from store", "Error", err.Error())
+		return p.cmdError(args.UserId, args.ChannelId, "Something went wrong.")
+	}
+
+	if len(links) == 0 {
+		return p.cmdError(args.UserId, args.ChannelId, "No links present.")
+	}
+
+	p.sendBotEphemeralPost(args.UserId, args.ChannelId, commandWaitingMessage)
+	go p.SendLinksWithDetails(args.UserId, args.ChannelId, links)
+	return &model.CommandResponse{}, nil
+}
+
+func (p *Plugin) SendLinksWithDetails(userID, channelID string, links []*storemodels.ChannelLink) {
+	var sb strings.Builder
+	sb.WriteString("| Mattermost Team | Mattermost Channel | MS Teams Team | MS Teams Channel | \n| :------|:--------|:-------|:-----------|")
+	errorsFound := false
+	wg := sync.WaitGroup{}
+	batchSize := 10
+	mattermostChannelIDsVsNames := make(map[string]string)
+	mattermostTeamIDsVsNames := make(map[string]string)
+
+	msTeamsTeamIDsVsNames := make(map[string]string)
+	msTeamsChannelIDsVsNames := make(map[string]string)
+	msTeamsTeamIDsVsChannelsQuery := make(map[string]string)
+
+	// Process all the links in batches using goroutines
+	for idx := 0; idx < len(links); idx += batchSize {
```

Comment:
This is generating 1 goroutine per batch of 10 elements, that means that if you have 10.000 links, you are spawing 1000 goroutines that are going to hammer our database with 20 requests each. That means, 20.000 DB request in parallel. That is crazy. I would try to move this directly to the store, creating a new store method like "LinksWithNames", that returns a new links structures that contains the channel name and team name using a database Join. We can even do that for all the regular queries of links, probably is not going to have a performance impact, and we can change it later if that is the case.

That way we can get rid of all the store queries here, and the extra complexity around goroutines and syncrhonization.